x/mint/abci_app.go

```diff
@@ -12,15 +12,15 @@ func BeginBlocker(ctx sdk.Context, k Keeper) {
 	params := k.GetParams(ctx)
 
 	// recalculate inflation rate
-	totalSupply := k.sk.TotalTokens(ctx)
+	stakingSupply := k.sk.StakingTokenSupply(ctx)
 	bondedRatio := k.sk.BondedRatio(ctx)
 	minter.Inflation = minter.NextInflationRate(params, bondedRatio)
-	minter.AnnualProvisions = minter.NextAnnualProvisions(params, totalSupply)
+	minter.AnnualProvisions = minter.NextAnnualProvisions(params, stakingSupply)
 	k.SetMinter(ctx, minter)
 
 	// mint coins, add to collected fees, update supply
 	mintedCoin := minter.BlockProvision(params)
 	k.fck.AddCollectedFees(ctx, sdk.Coins{mintedCoin})
```

Comment:
I'm starting to feel like `AddCollectedFees` is a misnomer and maybe even the `FeeCollectionKeeper`. A discussion for a separate issue/PR though.

cc @rigelrozanski

Reply:
yeah let's hear it in a new issue bex (please tag me)